x/oracle/keeper/keeper.go

```diff
@@ -60,15 +78,19 @@ func (k Keeper) GetProphecy(ctx sdk.Context, id string) (types.Prophecy, bool) {
 	return deSerializedProphecy, true
 }
 
-// setProphecy saves a prophecy with an initial claim
-func (k Keeper) setProphecy(ctx sdk.Context, prophecy types.Prophecy) {
-	store := ctx.KVStore(k.storeKey)
-	serializedProphecy, err := prophecy.SerializeForDB()
+// SetProphecy saves a prophecy with an initial claim
+func (k Keeper) SetProphecy(ctx sdk.Context, prophecy types.Prophecy) {
+	dbProphecy, err := prophecy.SerializeForDB()
```

Comment:
I know this is older code ,but it might make sense to return the error here , instead of panic. We can always panic at a higher level if needed , or just fail the transaction if possible .